test/stub.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * This file was heavily based by tests of mod_auth_openidc, which has the following
+ * notice:
+ *
+ * Copyright (C) 2017-2019 ZmartZone IAM
+ * Copyright (C) 2013-2017 Ping Identity Corporation
+ * All rights reserved.
```

Comment:
What is the actual license?
If there isn't an explicit grant of rights, then all that remains is this "all rights reserved" which means you can't use or distribute this code.

Reply:
mod_auth_openidc as a whole is licensed under Apache 2.0.
The stub.c file has no license blob on its own, unfortunately (https://github.com/zmartzone/mod_auth_openidc/blob/master/test/stub.c) but the tests.c  file (https://github.com/zmartzone/mod_auth_openidc/blob/master/test/test.c has Apache license blobs).

What explicit grant of rights should we look for? Maybe we could ask mod_auth_openidc upstream to add a note?

Since the tests are a separate binary, could the binary itself use the Apache license?

Reply:
This was resolved in another branch: https://github.com/jhrozek/mod_auth_mellon/tree/redirect_hardening
I'll open a PR with the additional hardening once this gets fixed.